meson.build

```diff
@@ -251,6 +250,8 @@ py.extension_module(
 )
 
 if get_option('tests').enabled()
+    libm = cc.find_library('m')
+
```

Comment:
This is a change for the sake of change. Please revert.

Reply:
I't not a "change for the sake of change", it's a change doesn't affect unix user and affect windown users. we should not even try to find libm on windows.

Reply:
Why moving looking up the `libm` within the `if get_option('tests').enabled()` section solves this problem?

Reply:
yes it doens't, I have fixed this. see latest diff

Reply:
If this is required, it is a Meson regression. Meson has special handling of `libm` when compiling with MSVC. Can you please share verbose logs of the build failing?
```sh
python -m pip install -v .
```

Reply:
Meson has special handling of libm, which is somehow broken.

I have tried `meson setup build --wipe -Dbuildtype=release` with CC=clang, CC=clang-cl and `meson setup build --wipe -Dbuildtype=release --backend vs` and they all failed with `meson.build:6:10: ERROR: C shared or static library 'm' not found`, only working compiler is GCC on windows.

```text
Collecting meson>=1.2.1
    Obtaining dependency information for meson>=1.2.1 from https://files.pythonhosted.org/packages/76/73/3dc4edc855c9988ff05ea5590f5c7bda72b6e0d138b2ddc1fab92a1f242f/meson-1.6.0-py3-none-any.whl.metadata
    Using cached meson-1.6.0-py3-none-any.whl.metadata (1.8 kB)
  Collecting packaging>=19.0 (from meson-python>=0.14.0)
    Obtaining dependency information for packaging>=19.0 from https://files.pythonhosted.org/packages/88/ef/eb23f262cca3c0c4eb7ab1933c3b1f03d021f2c48f54763065b6f0e321be/packaging-24.2-py3-none-any.whl.metadata
    Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
  Collecting pyproject-metadata>=0.7.1 (from meson-python>=0.14.0)
    Obtaining dependency information for pyproject-metadata>=0.7.1 from https://files.pythonhosted.org/packages/e8/61/9dd3e68d2b6aa40a5fc678662919be3c3a7bf22cba5a6b4437619b77e156/pyproject_metadata-0.9.0-py3-none-any.whl.metadata
    Using cached pyproject_metadata-0.9.0-py3-none-any.whl.metadata (6.9 kB)
  Using cached meson_python-0.17.1-py3-none-any.whl (27 kB)
  Using cached meson-1.6.0-py3-none-any.whl (973 kB)
  Using cached packaging-24.2-py3-none-any.whl (65 kB)
  Using cached pyproject_metadata-0.9.0-py3-none-any.whl (18 kB)
  Installing collected packages: packaging, meson, pyproject-metadata, meson-python
    Creating C:\Users\Trim21\AppData\Local\Temp\pip-build-env-3oiubf7d\overlay\Scripts
  Successfully installed meson-1.6.0 meson-python-0.17.1 packaging-24.2 pyproject-metadata-0.9.0
  Installing build dependencies ... done
  Running command Getting requirements to build wheel
  Getting requirements to build wheel ... done
  Running command Preparing metadata (pyproject.toml)
  + meson setup C:\Users\Trim21\proj\beancount C:\Users\Trim21\proj\beancount\.mesonpy-_3dwmx4z -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=C:\Users\Trim21\proj\beancount\.mesonpy-_3dwmx4z\meson-python-native-file.ini
  The Meson build system
  Version: 1.6.0
  Source dir: C:\Users\Trim21\proj\beancount
  Build dir: C:\Users\Trim21\proj\beancount\.mesonpy-_3dwmx4z
  Build type: native build
  WARNING: Project does not target a minimum version but uses feature introduced in '1.1': meson.options file. Use meson_options.txt instead
  Project name: beancount
  Project version: 3.1.0.dev0
  C compiler for the host machine: sccache clang (clang 19.1.3 "clang version 19.1.3")
  C linker for the host machine: clang link 14.41.34123.0
  Host machine cpu family: x86_64
  Host machine cpu: x86_64
  Program bison found: YES 3.8.2 3.8.2 (C:\Users\Trim21\scoop\shims\bison.EXE)
  Program flex found: YES 2.6.4 2.6.4 (C:\Users\Trim21\scoop\shims\flex.EXE)

  ..\meson.build:6:10: ERROR: C shared or static library 'm' not found
```

Reply:
Ah. `required: false` is missing. Just change `libm = cc.find_library('m')` to `libm = cc.find_library('m', required: false)`

Reply:
yes, I already did that.

And I think nowadays even on posix libm is linked by default. I remove libm and compile it with gcc 12, everything is fine.